components/brave_wallet_ui/components/desktop/views/portfolio/portfolio-asset.tsx

```diff
@@ -535,6 +536,13 @@ export const PortfolioAsset = (props: Props) => {
       }
       sendMessageToNftUiFrame(nftDetailsRef.current.contentWindow, command)
     }
+
+    // check if selectedAsset has an icon
+    if (selectedAsset && nftMetadata && stripERC20TokenImageURL(selectedAsset.logo) === '') {
+      const updatedAsset = { ...selectedAsset, logo: nftMetadata?.imageURL || '' }
+      dispatch(WalletActions.removeUserAsset(selectedAsset))
+      dispatch(WalletActions.addUserAsset(updatedAsset))
```

Comment:
We should add an updateUserAsset action to prevent updating the store twice

Reply:
Fixed 574e2cc547881632efbaec9513fe3689ab21576b